src/material.cc

```diff
@@ -15,15 +15,17 @@ const ResourceType Material::kType = "Material";
 Material::~Material() {}
 
 Material::Ptr Material::Create(Agent* creator, double quantity,
-                               Composition::Ptr c) {
-  Material::Ptr m(new Material(creator->context(), quantity, c));
+                               Composition::Ptr c, int package_id) {
+  Material::Ptr m(new Material(creator->context(), quantity, c, package_id));
   m->tracker_.Create(creator);
   return m;
 }
 
 Material::Ptr Material::CreateUntracked(double quantity,
                                         Composition::Ptr c) {
-  Material::Ptr m(new Material(NULL, quantity, c));
+  // default package id for untracked
+  int package_id = 1;
+  Material::Ptr m(new Material(NULL, quantity, c, package_id));
```

Comment:
Probably don't need this change if you add a default in the constructor declaration.